cloud/cluster/node_update.go

```diff
@@ -0,0 +1,39 @@
+package cluster
+
+import (
+	"fmt"
+)
+
+// NodeUpdate represents a change to the cluster
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+// Helper functions to create NodeUpdates
+
+func NewAdd(node Node) NodeUpdate {
+	return NodeUpdate{
+		NodeAdded,
+		node.Id(),
+		node,
+	}
+}
+
+func NewRemove(id NodeId) NodeUpdate {
+	return NodeUpdate{
+		UpdateType: NodeRemoved,
+		Id:         id,
+	}
+}
+
+type NodeUpdateSorter []NodeUpdate
```

Comment:
I believe []NodeUpdate should accurately represent a stream of updates. Can you rework the tests to handle default ordering? It's deterministic so it should be doable. Is there something else going on here that you're trying to address?